### PR TITLE
pd_worker, split_controller: introduce the new config and CPU collector registration mechanism

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: ref #12063. This is a sub-PR of #12593.

What's Changed:

```commit-message
Introduce the new split config and CPU collector registration mechanism.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: components/raftstore/src/store/worker/split_config.rs [hunk 1]
```diff
@@ -19,6 +19,16 @@ const DEFAULT_SPLIT_BALANCE_SCORE: f64 = 0.25;
 // We get contained score by sample.contained/(sample.right+sample.left+sample.contained). It will be used to avoid to split regions requested by range.
 const DEFAULT_SPLIT_CONTAINED_SCORE: f64 = 0.5;
 
+// When the gRPC poll thread CPU usage is higher than gRPC poll thread count * `DEFAULT_GRPC_THREAD_CPU_OVERLOAD_THRESHOLD_RATIO`,
+// the CPU-based split won't be triggered to prevent from increasing the CPU usage.
+const DEFAULT_GRPC_THREAD_CPU_OVERLOAD_THRESHOLD_RATIO: f64 = 0.5;
+// When the Unified Read Poll thread CPU usage is higher than Unified Read Poll thread count * `DEFAULT_UNIFIED_READ_POOL_THREAD_CPU_OVERLOAD_THRESHOLD_RATIO`,
+// the CPU-based split will be triggered to split the top hot CPU region.
```
Comment (rleungx) [inline components/raftstore/src/store/worker/split_config.rs:26]: I suggest explaining the priority of these parameters.

Comment (JmPotato) [reply components/raftstore/src/store/worker/split_config.rs:26]: What do you mean by "priority"? I think these thresholds don't quite have a term on priority.

Comment (rleungx) [reply components/raftstore/src/store/worker/split_config.rs:26]: I mean the case when `DEFAULT_GRPC_THREAD_CPU_OVERLOAD_THRESHOLD_RATIO` is more than 0.5 and `DEFAULT_UNIFIED_READ_POOL_THREAD_CPU_OVERLOAD_THRESHOLD_RATIO` is more than 0.8.

File: components/raftstore/src/store/worker/pd.rs [hunk 2]
```diff
@@ -1924,6 +1969,8 @@ where
             } => self.handle_update_max_timestamp(region_id, initial_status, txn_ext),
             Task::QueryRegionLeader { region_id } => self.handle_query_region_leader(region_id),
             Task::UpdateSlowScore { id, duration } => self.slow_score.record(id, duration.sum()),
+            Task::RegisterRegionCPUCollector => self.handle_register_region_cpu_collector(),
+            Task::DeregisterRegionCPUCollector => self.handle_deregister_region_cpu_collector(),
```
Comment (rleungx) [inline components/raftstore/src/store/worker/pd.rs:1973]: How about merging these two tasks?

Comment (JmPotato) [reply components/raftstore/src/store/worker/pd.rs]: Done. PTAL.

**Review by rleungx — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**